backend/src/index.ts

```diff
@@ -25,12 +26,14 @@ cloudinary.config({
 });
 
 const app = express();
+app.use('/api/order/checkout/webhook', express.raw({ type: 'application/json' }));
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/46#issuecomment-2428558658>#af29df</a><span> - 10/22/2024, 08:14 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The webhook endpoint is set up, but there's no visible implementation of Stripe signature verification for security.
</div>
<div id="fix">
<b>Fix:</b> Implement Stripe signature verification for the webhook endpoint using the Stripe library and a securely stored webhook secret.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -1,4 +1,5 @@
  import express from 'express';
 +import Stripe from 'stripe';
  import cors from 'cors';
  import mongoose from 'mongoose';
  import dotenv from 'dotenv';
 @@ -28,0 +30,14 @@
 +const stripe = new Stripe(process.env.STRIPE_SECRET_KEY as string, { apiVersion: '2022-11-15' });
 +
 +app.post('/api/order/checkout/webhook', express.raw({type: 'application/json'}), (req, res) => {
 +  const sig = req.headers['stripe-signature'] as string;
 +  let event;
 +  try {
 +    event = stripe.webhooks.constructEvent(req.body, sig, process.env.STRIPE_WEBHOOK_SECRET as string);
 +  } catch (err) {
 +    console.error('Webhook signature verification failed:', err.message);
 +    return res.sendStatus(400);
 +  }
 +  // Handle the event
 +});
 -app.use('/api/order/checkout/webhook', express.raw({ type: 'application/json' }));
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged